integration/cmd/fs/cp_test.go

```diff
@@ -36,22 +36,22 @@ func setupSourceFile(t *testing.T, ctx context.Context, f filer.Filer) {
 }
 
 func assertTargetFile(t *testing.T, ctx context.Context, f filer.Filer, relPath string) {
-	var err error
-
-	r, err := f.Read(ctx, relPath)
-	assert.NoError(t, err)
-	defer r.Close()
-	b, err := io.ReadAll(r)
-	require.NoError(t, err)
-	assert.Equal(t, "abc", string(b))
+	assertFileContent(t, ctx, f, relPath, "abc")
 }
 
 func assertFileContent(t *testing.T, ctx context.Context, f filer.Filer, path, expectedContent string) {
 	r, err := f.Read(ctx, path)
-	require.NoError(t, err)
+	if !assert.NoError(t, err) {
+		return
+	}
+
 	defer r.Close()
+
 	b, err := io.ReadAll(r)
-	require.NoError(t, err)
+	if !assert.NoError(t, err) {
```

Comment:
Double assertion  you need to know what NoError returns exactly makes it hard to read. Perhaps can be rewritten with condition on err + assert?